pkg/api/addons.go

```diff
@@ -152,7 +160,7 @@ func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
 
 	defaultReschedulerAddonsConfig := KubernetesAddon{
 		Name:    ReschedulerAddonName,
-		Enabled: to.BoolPtr(DefaultReschedulerAddonEnabled),
+		Enabled: to.BoolPtr(DefaultReschedulerAddonEnabled && !cs.Properties.IsAzureStackCloud()),
```

Comment:
Carried this additional default gate over from the "last mile" "concatenate addons string" business logic area.